python/cudf/cudf/core/multiindex.py

```diff
@@ -481,7 +476,8 @@ def copy(
 
         mi = MultiIndex._from_data(self._data.copy(deep=deep))
         if self._levels is not None:
-            mi._levels = [s.copy(deep) for s in self._levels]
+            # TODO: why is deep=True not being respected here
+            mi._levels = [idx.copy(deep) for idx in self._levels]
```

Comment:
@wence- could use some guidance here regarding the failing test below: `test_multiindex_copy_deep` for the `deep=True` case

Before `self._levels` was a list of `Series`, so it was performing a `Series.copy(True)` so the pointer checking in the test was passing. With this PR, now `self._levels` is a list of `Index`, and a `Index.copy(True)` is not passing the pointer checking in the test.

`deep=True` looks to be passed down to `Column.copy` and friends so I am not sure why the pointer in this case is not changing

Reply:
I'll take a look, thanks

Reply:
```suggestion
            mi._levels = [idx.copy(deep=deep) for idx in self._levels]
```

`idx.copy` takes a bunch more arguments (same as pandas) than just `deep=True/False`.

Reply:
🤦 thanks for pointing out my oversight!